### PR TITLE
Automate chores after releasing a new version

### PR DESCRIPTION
**Pull request by waiting-for-dev**
## Summary

After automating the setup for a new release in #4791, this PR adds
another GitHub action to streamline the chores that need to be done
after that.

We're adding the "release-generate" label to the PR generated after
setting up for a release. When merged, PRs with that label will take
care of triggering the new automation:

- Bumping the just-released version number to the `.dev` stage of the next
one.
- Bumping the versions on `docker-compose.yml` and
  `Spree.previous_solidus_minor_version` when the base branch is master.
- Publishing the matching release notes and creating the corresponding
  tag.
- Creating a new branch when the base branch is master.

When we set up for a new release on master, we need to have in mind
whether the next one after that will be another minor or a major. That
will change the development version we need to bump to. Because of that,
we're adding a new boolean parameter to the manually dispatchable action
to set up the release. When we specify that the next one is going to be
a major, we also add the `release-generate-last_minor` label to the new
PR and use it to calculate different version targets.

We're updating our conventions so that all branches point to the `.dev`
version of the next release. Only the last commits to be included in the
releases got that suffix removed.

## Checklist

Check out our [PR guidelines](https://github.com/solidusio/.github/blob/master/CONTRIBUTING.md#pull-request-guidelines) for more details.

The following are mandatory for all PRs:

- [x] I have written a thorough PR description.
- [x] I have kept my commits small and atomic.
- [x] I have used clear, explanatory commit messages.

The following are not always needed (~cross them out~ if they are not):

- ~[ ] I have added automated tests to cover my changes.~
- ~[ ] I have attached screenshots to demo visual changes.~
- ~[ ] I have opened a PR to update the [guides](https://github.com/solidusio/edgeguides).~
- ~[ ] I have updated the README to account for my changes.~

### CODE REVIEW

**Review by elia — APPROVED** [review]
As for the other PRs in this series I'm a bit worried about maintaining this code, but being very self contained and isolated we're not trapping ourselves too much, kudos for taking up the challenge @waiting-for-dev 🙌

**Comment by kennyadsl** [comment]
Thanks Marc, a couple of things:

> We're updating our conventions so that all branches point to the .dev
version of the next release. Only the last commits to be included in the
releases got that suffix removed.

We can start doing this change immediately, right?

---

Do you think it might be beneficial to create some diagrams to document the various steps of this flow? I think it could be handy in the future when we actually need to use this workflow and we don't want to read it via the code directly to understand what is about to happen.

File: .github/workflows/prepare_post_release.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+          message: |
+            Post-release chores after having released Solidus ${{ needs.extract_pipeline_context.outputs.candidate_tag }}
+            
+            
+            This code has been automatically generated by our 'Prepare post-release' GitHub
+            action.
+            
+            Make sure that:
+            
+            - The new release has been published, along with its tag. See https://github.com/${{ github.repository }}/releases/tag/${{ needs.extract_pipeline_context.outputs.candidate_tag }}
```
[the remark this answers is not in the document]
Comment (waiting-for-dev) [reply .github/workflows/prepare_post_release.yml]: Makes sense. Done!

**Comment by waiting-for-dev** [comment]
> As for the other PRs in this series I'm a bit worried about maintaining this code, but being very self contained and isolated we're not trapping ourselves too much, kudos for taking up the challenge @waiting-for-dev 🙌

@elia, I hear you. Yeah, as you said, we could just ditch the whole folder and keep doing everything manually. That doesn't affect our domain model in any way, but I'd also love to have more time and think of a creative way to maximize test coverage.

> We can start doing this change immediately, right?

Yeah, we can manually change them now, or wait for the first round of automation. It'll work either way.

> Do you think it might be beneficial to create some diagrams to document the various steps of this flow? I think it could be handy in the future when we actually need to use this workflow and we don't want to read it via the code directly to understand what is about to happen.

Absolutely!! That's the next task in my list 🙂

**Review by kennyadsl — APPROVED** [review]

**Pull request merged**